### PR TITLE
Add deploy section into .travis.yml

### PR DESCRIPTION
**Pull request by Jamim**
Hello @carlosalberto and @yurishkuro,

The main purpose of this PR is to simplify the deployment flow.
Please note that user and encrypted password must be set properly.

Useful links:
 - [docs.travis-ci.com/user/encryption-keys](https://docs.travis-ci.com/user/encryption-keys)
 - [docs.travis-ci.com/user/deployment/pypi](https://docs.travis-ci.com/user/deployment/pypi)
 - [docs.travis-ci.com/user/deployment#examples-of-conditional-deployment](https://docs.travis-ci.com/user/deployment#examples-of-conditional-deployment)
 - [docs.travis-ci.com/user/conditional-builds-stages-jobs](https://docs.travis-ci.com/user/conditional-builds-stages-jobs)
 - [docs.travis-ci.com/user/conditions-v1](https://docs.travis-ci.com/user/conditions-v1)

Best regards!

### CODE REVIEW

**Comment by Jamim** [comment]
Hello @yurishkuro,
Can you please take a look at this PR?
It's similar to uber-common/opentracing-python-instrumentation#94
Thank you!

**Comment by yurishkuro** [comment]
@Jamim I don't maintain this repo, nor do I have access to its pypi package, so I can't create the secrets.

cc @opentracing-contrib/ot-contrib-python-maintainers

**Comment by Jamim** [comment]
Hello @carlosalberto,
Could you please take a look at this PR?
Thanks!

**Comment by Jamim** [comment]
ping @carlosalberto

File: .travis.yml [hunk 1]
```diff
@@ -13,12 +13,19 @@ python:
 stages:
   - lint
   - test
+  - name: deploy
+    if: tag is present
```
Comment (carlosalberto) [inline .travis.yml]: If I understand correctly, this is done so a new tag will cause a deployment, right?

File: .travis.yml [hunk 2]
```diff
@@ -13,12 +13,19 @@ python:
...
 
 matrix:
   include:
     - stage: lint
       name: flake8
       python: '3.7'
+    - stage: deploy
+      name: Deploy to PyPI
+      python: '3.7'
```
Comment (carlosalberto) [inline .travis.yml]: Do we really need Python 3.7 for deploying? Just curious about why we don't use Python 2.7 ;)

File: .travis.yml (hunk 1 above)
Comment (Jamim) [reply .travis.yml]: Hello @carlosalberto,
Right. Any new tag pushed to GitHub will cause a deployment.

File: .travis.yml (hunk 2 above)
Comment (Jamim) [reply .travis.yml]: I suppose that any Python version could be used for deploying.
At the same time, I believe that it makes sense to use a version which [will be EOLed](https://devguide.python.org/#status-of-python-branches) later than sooner, so Python 2.7 with end-of-life on 2020-01-01 is not a good option.

**Pull request merged**